marathon/marathon.go

```diff
@@ -32,7 +36,7 @@ type LeaderResponse struct {
 	Leader string `json:"leader"`
 }
 
-func New(config Config) (*Marathon, error) {
+func New(config Config, leader string) (*Marathon, error) {
```

Comment:
I think we should move sync leader from sync.Config to marathon config and replace [`marathon.Marathoner.Leader()`](https://github.com/allegro/marathon-consul/blob/master/marathon/marathon.go#L2) with `AmILeader` and use it in sync. This could be done in separate PR.